### ISSUE TITLE
Stuck at snapshotting caffemodel

### ISSUE THREAD

**Issue by iN1k1** [BVLC/caffe#3662]
HI All,
I'm facing a weird problem when caffe is trying to snapshotting the caffemodel on disk.
It is only able to partially save the model, then it suddenly stops and hangs forever.
I checked the HDD where the data is being saved. No corrupted sectors or other issues..
The only way to use caffe again is to reboot the server...

Thanks!

**Comment by seanbell** [comment]
Please read [CONTRIBUTING.md](https://raw.githubusercontent.com/BVLC/caffe/master/CONTRIBUTING.md) before filing bugs:
> When reporting a bug, it's most helpful to provide the following information, where applicable:
> 
> * What steps reproduce the bug?
* Can you reproduce the bug using the latest [master](https://github.com/BVLC/caffe/tree/master), compiled with the `DEBUG` make option?
* What hardware and operating system/distribution are you running?
* If the bug is a crash, provide the backtrace (usually printed by Caffe; always obtainable with `gdb`).

This sounds like a filesystem problem -- are you out of space?  Is the filesystem still usable?  Is this an NFS?  You have not given enough info to help; we cannot read your mind.

**Comment by SvenTwo** [comment]
I have the same issue with hanging processes during the snapshot sometimes (Saving onto local hard drives; four magnetic drives; software raid0; ext4, Ubuntu 14 LTS). It's a lot of disk io particularly if multiple instances are running in parallel; maybe it's just a deficit of the kernel when you buffer tons of data.

I cannot reproduce it right now, but maybe replacing the binary writing using the protobuf saving buffers can help. E.g.:

```
void WriteProtoToBinaryFile(const Message& proto, const char* filename) {
  int fd = open(filename, O_WRONLY | O_CREAT | O_TRUNC, 0644);
  FileOutputStream* raw_output = new FileOutputStream(fd);
  CodedOutputStream *coded_output = new CodedOutputStream(raw_output);
  CHECK(proto.SerializeToCodedStream(coded_output));
  delete coded_output;
  delete raw_output;
  close(fd);
}
```

**Comment by iN1k1** [comment]
Sorry Sean, my apologizes. 
I have a similar configuration to Sven. Ubuntu 14.04 LTS, 1TB NTFS formatted SSD, raid 0. However, I'm facing the problem even only one single caffe instance is running. I'll try to reproduce the error and report with the DEBUG make option.
I also saw that once the error appears the system produces several kernel logs reporting that a process is not responding in 120sec. It seems that the process is stuck in the IO and is owned by the kernel, because I also cannot kill it.

**Comment by iN1k1** [comment]
Hi all,
I actually solved the issue by formatting the HDD to ext4. Whenever I have time, I'll try to reproduce the issue with another drive to see whether the issue was due to the OS or Caffe. Even I feel that the problem was the OS..

**Comment by litingfeng** [comment]
I encountered the same issue, but my situation is that the HDD is already ext4(actually it says ext3/ext4 in property). I used the same solver file with two different train_net, however, only one of them could be snapshotted correctly, the other stuck in io. Can anybody give me some hints?
Thanks a lot.

**Comment by litingfeng** [comment]
Just now I found a solution: resize the input image to a smaller size.

**Comment by Noiredd** [comment]
Closing as solved. If this behavior reoccurs, I'll ask that a new issue be open with a detailed report.
